qiskit/circuit/qpy_serialization.py

```diff
@@ -778,6 +831,9 @@ def _read_instruction(file_obj, circuit, registers, custom_instructions):
         elif type_str == "e":
             container = io.BytesIO(data)
             param = _read_parameter_expression(container)
+        elif type_str == "v":
+            container = io.BytesIO(data)
+            param = _read_parameter_vec(container, vectors)
```

Comment:
I thought about this, but I'm thinking we might be ok without doing. Since we should be covered by the name/uuid uniqueness still being preserved here as a normal `Parameter`. For places where the parameter for phase is also in the circuit as a gate parameter we'll be covered by the gate's serialization. The only case where we'd lose information is if there was a standalone `ParameterVectorElement` for global phase, something like:

```
vec = ParameterVector(phase, 1)
qc = QuantumCircuit(2, global_phase=vec[0])
```

where the vec context would be lost. But I wasn't sure it was worth worrying about that or not. Either way it's a simple to fix we just move the vectors initialization to the top of `_read_circuit` and then add this block to `_read_header_v2` and similarly for writing of global phase in write_circuit

Reply:
I went ahead and did this in: https://github.com/Qiskit/qiskit-terra/pull/7381/commits/2ec1ee4a41bcfd6b9be36e85c194132a423d6a6a since we already need to pass `vectors` to `_read_header_v2` for parameter expressions that contain vector elements and it was just 2 lines to support it.